312 plugin isolated classloader

delbertlegg opened the pull request:
Fix for #312

P3trur0 commented on framework/freedomotic-core/src/main/java/com/freedomotic/plugins/PluginIsolatedClassloader.java:
```diff
@@ -60,7 +65,7 @@ public boolean accept(File pathname) {
                     urls[i] = jarFiles[i].toURI().toURL();
                 } catch (MalformedURLException e) {
-                    throw new RuntimeException(
+                    throw new FreedomoticException(
```
Please use FreedomoticRuntimeException here

P3trur0 commented on framework/freedomotic-core/src/main/java/com/freedomotic/plugins/PluginIsolatedClassloader.java:
```diff
@@ -34,23 +36,26 @@
-                return pathname.getName().endsWith(".jar");
-            }
-        };
+        FileFilter jarFilter = (File pathname) -> pathname.getName().endsWith(".jar");
```
Could this lambda be written directly in the method call at line 60?

mcicolella commented:
In this PR I see a commit also present in #316 . Please don't mix up issues.

delbertlegg commented:
I'll make the mentioned updates. Thanks!

delbertlegg commented:
Updated per comments. Thanks!

P3trur0 commented on framework/freedomotic-core/src/main/java/com/freedomotic/plugins/PluginIsolatedClassloader.java:
```diff
@@ -53,20 +54,19 @@
      */
     public PluginIsolatedClassloader(String jarDir) throws FreedomoticException {
```
This `throws FreedomoticException` should be removed since `FreedomoticRuntimeException` is unchecked.

P3trur0 commented:
@delbertlegg good! See my last comment too, please :) Then for me that's totally ok.

delbertlegg commented:
@P3trur0 Yes sir...got that in the second commit there. Should be good to go!

P3trur0 commented:
@delbertlegg you're right, sorry :+1:

mcicolella commented:
@delbertlegg can you give a rebase to your branch to have a single commit with the definitive changes? Thanks

delbertlegg commented:
@P3trur0 No worries! Caught it after the initial commit and added after.
@mcicolella I rebased but I don't see any difference in the pull request (sorry, more familiar with SVN than git at this point). Do you see a change?

mcicolella commented:
Integrated! Thanks
